allauth/socialaccount/providers/shopify/views.py

```diff
@@ -0,0 +1,58 @@
+from allauth.socialaccount.providers.oauth2.client import OAuth2Client
+from allauth.socialaccount.providers.oauth2.views import (OAuth2Adapter,
+                                                          OAuth2LoginView,
+                                                          OAuth2CallbackView)
+import requests
+from .provider import ShopifyProvider
+
+
+class ShopifyOAuth2Client(OAuth2Client):
+    def get_redirect_url(self, request, extra_params):
+        self.shop = extra_params['shop']
+        _url = 'https://{}.{}'.format(
+            self.shop,
+            super(ShopifyOAuth2Client, self).get_redirect_url(request, extra_params))
+        return _url
+
+
+class ShopifyOAuth2LoginView(OAuth2LoginView):
+    def get_client(self, request, app):
+        client = super(ShopifyOAuth2LoginView, self).get_client(request, app)
+        client.__class__ = ShopifyOAuth2Client
+        return client
+
+
+class ShopifyOAuth2CallbackView(OAuth2CallbackView):
+    def get_client(self, request, app):
+        client = super(ShopifyOAuth2CallbackView, self).get_client(request, app)
+        client.access_token_url = 'https://{}/{}'.format(
+            request.GET.get('shop'), client.access_token_url)
```

Comment:
Wondering: I am not fully familiar with shopify, but are only subdomains allowed here? How about shop owners that have their own fqdn ?